testing/trino-benchto-benchmarks/src/test/java/io/trino/sql/planner/statistics/GlueStatisticsGenerator.java

```diff
@@ -0,0 +1,148 @@
+import static io.trino.plugin.hive.HivePartition.UNPARTITIONED_ID;
+import static io.trino.plugin.hive.HiveTestUtils.SESSION;
+import static io.trino.plugin.hive.statistics.MetastoreHiveStatisticsProvider.getPartitionsStatistics;
+import static io.trino.spi.transaction.IsolationLevel.READ_UNCOMMITTED;
+import static io.trino.sql.planner.TestHiveTpcdsCostBasedPlan.TPCDS_SF1000_STATISTICS_REPOSITORY;
+import static io.trino.sql.planner.TestHiveTpchCostBasedPlan.TPCH_SF1000_STATISTICS_REPOSITORY;
+import static java.util.Objects.requireNonNull;
+
+public class GlueStatisticsGenerator
+        implements AutoCloseable
+{
+    private final HiveConnector connector;
+    private final ConnectorTransactionHandle transaction;
+    private final HiveMetadata metadata;
+    private final HivePartitionManager partitionManager;
+
+    public GlueStatisticsGenerator()
+    {
+        connector = getHiveConnector();
+        transaction = connector.beginTransaction(READ_UNCOMMITTED, true, true);
+        metadata = (HiveMetadata) connector.getRawMetadata(SESSION, transaction);
+        partitionManager = new HivePartitionManager(new HiveConfig());
+    }
+
+    public void generate(String schema, TableStatisticsDataRepository repository)
+    {
+        requireNonNull(schema, "schema is null");
+        requireNonNull(repository, "repository is null");
+
+        metadata.listTables(SESSION, Optional.of(schema)).forEach(table -> this.generate(table, repository));
+    }
+
+    private void generate(SchemaTableName tableName, TableStatisticsDataRepository repository)
+    {
+        HiveTableHandle tableHandle = metadata.getTableHandle(SESSION, tableName);
+        HivePartitionResult partitionResult = partitionManager.getPartitions(
+                metadata.getMetastore(),
+                tableHandle,
+                Constraint.alwaysTrue());
+        List<HivePartition> partitions = partitionManager.getPartitionsAsList(partitionResult);
+
+        Map<String, PartitionStatistics> statistics = getPartitionsStatistics(metadata.getMetastore(), tableName, partitions);
+        writeStatistics(tableHandle, statistics, repository);
+    }
+
+    private void writeStatistics(
+            HiveTableHandle tableHandle,
+            Map<String, PartitionStatistics> statistics,
+            TableStatisticsDataRepository repository)
+    {
+        if (statistics.containsKey(UNPARTITIONED_ID)) {
+            repository.write(tableHandle.getTableName(), statistics.get(UNPARTITIONED_ID));
+        }
+        else {
+            HiveColumnHandle columnHandle = getOnlyElement(tableHandle.getPartitionColumns());
+            repository.writePartitionStatistics(tableHandle.getTableName(), columnHandle.getName(), statistics);
+        }
+    }
+
+    private static HiveConnector getHiveConnector()
+    {
+        return (HiveConnector) getHiveConnectorFactory().create(
+                "hive",
+                ImmutableMap.of(
```

Comment:
I don't think we need to hard-code glue or the hive catalog config here, we can create a resource file similar to `testing/trino-server-dev/etc/catalog/hive.properties` and use `loadPropertiesFrom` to create properties map from it.